src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,1536 @@
+
+    k = k + 1
+    j = j + 1
+
+    if j == 1:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 0]))
+    elif j == 2 or j == 3 or j == 4:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+    elif j == 5 or j == 9 or j == 13:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+    else:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 3]))
+
+    temp = np.array(sorted_matrix[:, 2]) * np.array(sorted_matrix[:, 3])
+    sorted_matrix = np.column_stack((sorted_matrix, temp))
+
+    expectation_val = np.sum(temp)
+
+    return expectation_val
+
+
+def GST_measure_13(circuit, idx_basis_op, k, j, nshots, save_data=None):
+    import numpy as np
+
+    nqubits = 1
+    idx_basis = ["I", "X", "Y", "Z"]
+
+    result = circuit.execute(nshots=nshots)
+    counts = dict(result.frequencies(binary=True))
+
+    if save_data is not None:
+        import os
+
+        filedirectory = "GST_basisops_qibo_1qb/"
+        fname = "BasisOp%.2d (k=%.2d,j=%.2d) %s_basis.txt" % (
+            idx_basis_op,
+            k,
+            j,
+            idx_basis[j],
+        )
+        # If location does not exist, create location.
+        GST_basisops_qibo_1qb = "GST_basisops_qibo_1qb"
+        if not os.path.exists(GST_basisops_qibo_1qb):
+            os.makedirs(GST_basisops_qibo_1qb)
+
+        # Save counts into a matrix. 0th column: word; 1st column: count value
+        # --------------------------------------------------------------------
+        fout = filedirectory + fname
+        fo = open(fout, "w")
+        for v1, v2 in counts.items():
+            fo.write(str(v1) + "  " + str(v2) + "\n")
+        fo.close()
+
+    # Find value for Bjk tilde
+    # -------------------------
+    expectation_signs = np.array([[1, 1, 1, 1], [1, -1, -1, -1]])
+    matrix = np.array([[0, 0], [1, 0]])
+
+    for v1, v2 in counts.items():
+        # print(v1, v2)
+        row = int(v1, 2)
+        col = v2
+        matrix[row, 1] = col
+
+    sorted_matrix = sort_counts(matrix)
+    probs = sorted_matrix[:, 1] / np.sum(sorted_matrix[:, 1])
+    sorted_matrix = np.column_stack((sorted_matrix, probs))
+
+    k = k + 1
+    j = j + 1
+
+    if j == 1:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 0]))
+    elif j == 2 or j == 3 or j == 4:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+    elif j == 5 or j == 9 or j == 13:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+    else:
+        sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 3]))
+
+    temp = np.array(sorted_matrix[:, 2]) * np.array(sorted_matrix[:, 3])
+    sorted_matrix = np.column_stack((sorted_matrix, temp))
+
+    expectation_val = np.sum(temp)
+
+    return expectation_val
+
+
+def GST_measure_241(circuit, idx_basis_ops, k, j, nshots, save_data=None):
```

Comment:
So I have combined all the four functions ```GST_measure_*``` into a single ```GST_measure``` function. It will be visible in the next version of ```gate_set_tomography.py```.